File: packages/agent-mux/watch-wearos-app/app/src/main/kotlin/ai/a5c/amux/wear/state/Projections.kt
package ai.a5c.amux.wear.state

data class WearRunProjection(
  val runId: String,
  val agent: String,
  val status: String,
  val costUsd: Double,
  val hasPendingHook: Boolean,
)

object WearProjections {
  fun merge(current: List<WearRunProjection>, incoming: List<WearRunProjection>): List<WearRunProjection> {
    return (current + incoming)
      .associateBy { it.runId }
      .values
      .sortedBy { it.runId }
  }
}

File: packages/agent-mux/watch-watchos-app/AgentMuxWatchApp/State/Projections.swift
import Foundation

struct WatchRunProjection: Identifiable, Equatable {
  let id: String
  let agent: String
  let status: String
  let costUsd: Double
  let hasPendingHook: Bool
}

enum WatchProjections {
  static func merge(_ current: [WatchRunProjection], incoming: [WatchRunProjection]) -> [WatchRunProjection] {
    Dictionary(uniqueKeysWithValues: (current + incoming).map { ($0.id, $0) }).values.sorted { $0.id < $1.id }
  }
}
